analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs

```diff
@@ -84,6 +85,11 @@ protected override ProgramState PreProcess(ProgramState state, IPropertyReferenc
                 state = state.SetOperationValue(operation, value);
             }
         }
+        if (ArithmeticCalculator.PropertyReferenceConstraint(state, operation, symbol) is { } constraint)
+        {
+            state = state.SetOperationConstraint(operation, constraint);
+        }
+        state = CollectionTracker.ApplyConstraints(state, operation, symbol);
```

Comment:
Maybe just merge both methods and put both of them in `CollectionTracker`:
```suggestion
        state = CollectionTracker.LearnFrom(state, operation, symbol);
```